src/libpmemset/pmemset.c

```diff
@@ -447,18 +447,53 @@ pmemset_pmem2_config_init(struct pmem2_config *pmem2_cfg,
  */
 static int
 pmemset_create_reservation(struct pmem2_vm_reservation **pmem2_reserv,
-		size_t size)
+		struct pmem2_source *src, size_t size)
 {
+	size_t alignment;
+	int ret = pmem2_source_alignment(src, &alignment);
+	if (ret)
+		return ret;
+
+	/*
+	 * vm reservation aligns to the Mmap_align, source can have different
+	 * alignment than mapping alignment e.g. devdax
+	 */
+	size_t rsv_size = size;
+	if (alignment != Mmap_align)
```

Comment:
I think I'm missing something.

Wouldn't it be enough to just `rsv_size = ALIGN_UP(size, alignment)`?